src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.audit.listeners;
+
+import hudson.ExtensionList;
+import hudson.Extension;
+import jenkins.security.SecurityListener;
+import io.jenkins.plugins.audit.RequestContext;
+
+import java.io.IOException;
+import java.lang.UnsupportedOperationException;
+import java.util.logging.Logger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Instant;
+import java.util.logging.Level;
+import javax.annotation.Nonnull;
+
+import io.jenkins.plugins.audit.event.Login;
+import io.jenkins.plugins.audit.event.Logout;
+import org.apache.logging.log4j.audit.LogEventFactory;
+import org.apache.logging.log4j.core.util.NetUtils;
+import org.acegisecurity.userdetails.UserDetails;
+
+/**
+ * Listener notified of user login and logout events.
+ */
+@Extension
+public class UserLogListener extends SecurityListener {
+
+    private static final Logger LOGGER = Logger.getLogger(UserLogListener.class.getName());
+
+    /**
+     * Fired when a user was successfully authenticated using credentials. It could be password or any other credentials.
+     * This might be via the web UI, or via REST (using API token or Basic), or CLI (remoting, auth, ssh)
+     * or any other way plugins can propose.
+     *
+     * UserLogListener does not use this abstract class method, however, rather than leaving
+     * its implementation empty it is made to throw an exception if the method is ever called.
+     *
+     * @param details details of the newly authenticated user, such as name and groups.
+     * @throws UnsupportedOperationException if the method is called wrongly.
+     */
+    @Override
+    protected void authenticated(@Nonnull UserDetails details) {
+        throw new UnsupportedOperationException("This class does not use this method.");
```

Comment:
You should just make these methods no-ops. That way, when this listener is invoked, it doesn't cause any errors elsewhere. These methods can be safely ignored.

Reply:
alright :)